api/app.go

```diff
@@ -0,0 +1,122 @@
+// kubecos api
+// https://github.com/topfreegames/kubecos
+//
+// Licensed under the MIT license:
+// http://www.opensource.org/licenses/mit-license
+// Copyright © 2017 Top Free Games <backend@tfgco.com>
+
+package api
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/gorilla/mux"
+	"github.com/topfreegames/kubecos/kubecos-cli/errors"
+	"github.com/topfreegames/kubecos/kubecos-cli/metadata"
+	"github.com/topfreegames/kubecos/kubecos-cli/models"
+	runner "gopkg.in/mgutz/dat.v2/sqlx-runner"
+)
+
+//App is our API application
+type App struct {
+	Address       string
+	DB            runner.Connection
+	Debug         bool
+	Logger        logrus.FieldLogger
+	Router        *mux.Router
+	Server        *http.Server
+	ServerControl *models.ServerControl
+	Login         *models.Login
+}
+
+//NewApp ctor
+func NewApp(host string, port int, debug bool, logger logrus.FieldLogger, login *models.Login) (*App, error) {
+	a := &App{
+		Address: fmt.Sprintf("%s:%d", host, port),
```

Comment:
The CLI needs to be hardcoded since port and host are defined on Google API resources for callback url. 
On mystack-controller, for example, they are defined as arguments of start command.